bases/rsptx/interactives/runestone/activecode/js/acfactory.js

```diff
@@ -112,25 +112,29 @@ export default class ACFactory {
         var lang = eBookConfig.acDefaultLanguage
             ? eBookConfig.acDefaultLanguage
             : "python";
-        if (lang === "java" || lang === "cpp" || lang === "python3") {
```

Comment:
This statement  does not contain all of the languages supported by runestone.  Only the languages that require the backend JOBE server to compile.  So you will miss languages like javascript, sql, html. But you cannot just add them to the if statement or use the `.includes` statement with all of them.

Reply:
👌I wondered... 
I'll go ahead and revert that change then.
@bnmnetp 
Out of curiosity, with respect to my languageNames dictionary, will I have to handle all the other languages supported by runestone? If that is the case, then the dictionary suggestion might have to be extended to handle those as well

Reply:
Yes, the dictionary should handle all of the languages we support.

Reply:
Can you be kind enough to direct me where to find those? I can poke around but your suggestion will make my life easier😂

Reply:
I think you can discern most of them by looking at the names in that if statement, plus the files in the js folder.  activecode_xxx.js There may also be a comprehensive list in activecode.js which is the baseclass for all of those.

Reply:
Those are all the languages I could find. Hopefully, I am not missing any. 
One thing I wasn't sure is if the `eBookConfig.acDefaultLanguage` can ever be abbreviated as shown by this comment below. I decided to handle those cases either ways.

>    //   lang - to render labels in a style appropriate for other languages,
>     //          and to display the proper language in langDisplayDiv:
>     //          'py2' for Python 2, 'py3' for Python 3, 'js' for JavaScript, 'java' for Java,
>     //          'ts' for TypeScript, 'ruby' for Ruby, 'c' for C, 'cpp' for C++
>     //          'py3anaconda' for Python 3 with Anaconda

Reply:
Just make sure that you don't crash out if the language is not in the dictionary.  Make a default, like python, but maybe also have your message say 'xxx is a known language please report this'

Reply:
Hmm, doesn't make sense to default to python if language is not in the dictionary. Its like defaulting to python for `kotlin`, this might just create misunderstanding

 What about defaulting to the value of `lang` instead even if it's not a programming language e.g cpp. Then add that message 'xxx is a known language please report this' for clarity